File: vertx-lang-php-jphp/src/main/php/io/vertx/php/core/eventbus/EventBus.php
<?php 
namespace io\vertx\php\core\eventbus;
class EventBus
{
    
    private function __construct()
    {

    }
    /**
     * (callable interceptor)
     * @return EventBus
     * methodSize : 1
     */
    public function addInterceptor(/*1*/$arg0)
    {

    }
    /**
     * (string address)
     * @return MessageConsumer
     * (string address, callable handler)
     * @return MessageConsumer
     * methodSize : 2
     */
    public function consumer(/*2*/$arg0, $arg1)
    {

    }
    /**
     * ()
     * @return bool
     * methodSize : 1
     */
    public function isMetricsEnabled(/*0*/)
    {

    }
    /**
     * (string address)
     * @return MessageConsumer
     * (string address, callable handler)
     * @return MessageConsumer
     * methodSize : 2
     */
    public function localConsumer(/*2*/$arg0, $arg1)
    {

    }
    /**
     * (string address, object message)
     * @return EventBus
     * (string address, object message, DeliveryOptions options)
     * @return EventBus
     * methodSize : 2
     */
    public function publish(/*3*/$arg0, $arg1, $arg2)
    {

    }
    /**
     * (string address)
     * @return MessageProducer
     * (string address, DeliveryOptions options)
     * @return MessageProducer
     * methodSize : 2
     */
    public function publisher(/*2*/$arg0, $arg1)
    {

    }
    /**
     * (callable interceptor)
     * @return EventBus
     * methodSize : 1
     */
    public function removeInterceptor(/*1*/$arg0)
    {

    }
    /**
     * (string address, object message)
     * @return EventBus
     * (string address, object message, callable replyHandler)
     * @return EventBus
     * (string address, object message, DeliveryOptions options)
     * @return EventBus
     * (string address, object message, DeliveryOptions options, callable replyHandler)
     * @return EventBus
     * methodSize : 4
     */
    public function send(/*4*/$arg0, $arg1, $arg2, $arg3)
    {

    }
    /**
     * (string address)
     * @return MessageProducer
     * (string address, DeliveryOptions options)
     * @return MessageProducer
     * methodSize : 2
     */
    public function sender(/*2*/$arg0, $arg1)
    {

    }
}

File: vertx-lang-php-jphp/src/main/test/io/vertx/php/core/eventbus/EventBus.java
package io.vertx.php.core.eventbus;
import php.runtime.annotation.Reflection.Name;
import php.runtime.annotation.Reflection.Namespace;
import php.runtime.annotation.Reflection.Signature;
import io.vertx.lang.jphp.IWrapper;
import io.vertx.php.core.metrics.Measured;
import io.vertx.php.core.eventbus.SendContext;
import io.vertx.php.core.eventbus.Message;
import io.vertx.php.core.eventbus.MessageConsumer;
import io.vertx.php.core.eventbus.MessageProducer;
@Name("EventBus")
@Namespace("io\\vertx\\php\\core\\eventbus")
public interface EventBus<S extends io.vertx.core.eventbus.EventBus> extends IWrapper<S>{
    
    @Signature
    default Memory isMetricsEnabled(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().isMetricsEnabled()
    }

    @Signature
    default Memory send(Environment __env__, Memory... args) {
        this.getWrappedObject().send(address, message)
    }

    @Signature
    default Memory publish(Environment __env__, Memory... args) {
        this.getWrappedObject().publish(address, message)
    }

    @Signature
    default Memory consumer(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, MessageConsumerImpl::new, this.getWrappedObject().consumer(address)
    }

    @Signature
    default Memory localConsumer(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, MessageConsumerImpl::new, this.getWrappedObject().localConsumer(address)
    }

    @Signature
    default Memory sender(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, MessageProducerImpl::new, this.getWrappedObject().sender(address)
    }

    @Signature
    default Memory publisher(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, MessageProducerImpl::new, this.getWrappedObject().publisher(address)
    }

    @Signature
    default Memory addInterceptor(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, EventBusImpl::new, this.getWrappedObject().addInterceptor(interceptor)
    }

    @Signature
    default Memory removeInterceptor(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, EventBusImpl::new, this.getWrappedObject().removeInterceptor(interceptor)
    }

}
